src/router.ts

```diff
@@ -0,0 +1,40 @@
+import Router from 'koa-router';
+import glob from 'glob';
+import path from 'path';
+import Koa from 'koa';
+import { isFunction } from 'lodash';
+
+import { config } from './config';
+
+export const router = new Router();
+
+// This route serves as a health check. Should return the mongo db connection's status and other services' if any.
+router.get('/ping', (ctx: Koa.Context) => {
+  ctx.body = { status: 'OK' };
+});
+
+router.get('/', (ctx: Koa.Context) => {
+  ctx.body = { name: config.appName, version: config.version };
+});
+
+const api = new Router();
+
+// This relatively ugly code snippet looks through the v1 folder in /src/lib for all the router.ts files
+// to look at all the routes. Pretty handy as it allows to just create a folder `toto` with a router.ts to load all
+// its routes without extraneous configuration.
+glob.sync('./v1/**/router.ts', { cwd: './src/lib' }).forEach((routerPath) => {
```

Comment:
does it work for compiled js?

Reply:
Oh you're right this won't work for compiled JS since it's going to look for .ts files in the src folder...

Replacing the extension in the pattern, and using `__dirname` to make sure to look in the right directory should make it compatible with compiled JS.

```suggestion
glob.sync('./v1/**/router.+(js|ts)', { cwd: `${__dirname}/lib` }).forEach((routerPath) => {
```

Reply:
Opened an issue to solve this problem https://github.com/Faz95210/test-backend/issues/10

Reply:
Solved ✅